<?xml version="1.0"?>
<!--robot xmlns:xacro="http://wiki.ros.org/xacro" name="mia_hand"-->

<robot xmlns:xacro="http://wiki.ros.org/xacro" >

<!-- Gazebo.xacro v5.1 -->

 <xacro:macro name="mia_gazebo" params="hw_interface_thumb_fle:=VelocityJointInterface
                                        hw_interface_index_fle:=VelocityJointInterface
                                        hw_interface_mrl_fle:=VelocityJointInterface
                                        robotNamespace:=/mia_hand
                                        LoadGZPlugin:=true">

  <!-- ////////////////////////////////////////////////////////////////////////////////////
    Choose the right hardware interface to use for joints based on the controllers ////
   ////////////////////////////////////////////////////////////////////////////////////-->



  <!-- /////////////////////////////////////////////////
    Add transmission for every not fixed joint ////
   ////////////////////////////////////////////////////-->

  <xacro:include filename="$(find mia_hand_description)/urdf/inc/mia_hand.transmission.xacro" />

  <xacro:mia_hand_transmission_macro joint_name="j_thumb_opp"          trans_type="transmission_interface/SimpleTransmission"  hw_interface="PositionJointInterface"/> <!-- Fake transmission-->
  <xacro:mia_hand_transmission_macro joint_name="j_index_fle" 	trans_type="transmission_interface/SimpleTransmission"  hw_interface="${hw_interface_index_fle}"/>
  <xacro:mia_hand_transmission_macro joint_name="j_thumb_fle" 	trans_type="transmission_interface/SimpleTransmission"  hw_interface="${hw_interface_thumb_fle}"/>
  <xacro:mia_hand_transmission_macro joint_name="j_mrl_fle" 	        trans_type="transmission_interface/SimpleTransmission"  hw_interface="${hw_interface_mrl_fle}"/>
  <xacro:mia_hand_transmission_macro joint_name="j_ring_fle"	 	trans_type="transmission_interface/SimpleTransmission"  hw_interface="PositionJointInterface"/> <!-- Fake transmission-->
  <xacro:mia_hand_transmission_macro joint_name="j_little_fle" 	trans_type="transmission_interface/SimpleTransmission"  hw_interface="PositionJointInterface"/> <!-- Fake transmission-->

  <!-- ros_control plugin -->
  <xacro:if value="${LoadGZPlugin}">


  <gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      <robotNamespace>${robotNamespace}</robotNamespace>
      <!--robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType -->
        <robotSimType>mia/MiaHWSim</robotSimType>
    </plugin>
  </gazebo>

  </xacro:if>

 <!-- Gazebo-customization elements including colors see also http://wiki.ros.org/simulator_gazebo/Tutorials/ListOfMaterials-->
  <!-- Palm -->

  <gazebo reference="palm">
    <material>Gazebo/Black</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>False</selfCollide>
  </gazebo>

 <!-- Thumb -->

  <gazebo reference="thumb_opp">
    <material>Gazebo/White</material>
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>100000.0</mu1>
    <mu2>10000.0</mu2>
    <fdir1>1 0 0</fdir1>
    <selfCollide>False</selfCollide>
  </gazebo>

  <gazebo reference="thumb_sensor">
    <material>Gazebo/White</material>

    <selfCollide>False</selfCollide>
  </gazebo>

  <gazebo reference="thumb_fle">
    <material>Gazebo/White</material>
    <kp>1000000.0</kp>
    <kd>100.0</kd>
    <mu1>100000.0</mu1>
    <mu2>10000.0</mu2>
    <fdir1>1 0 0</fdir1>
    <selfCollide>True</selfCollide>
  </gazebo>

  <!-- Index -->

  <gazebo reference="index_sensor">
    <material>Gazebo/White</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>True</selfCollide>
  </gazebo>

  <gazebo reference="index_fle">
    <material>Gazebo/White</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>True</selfCollide>
  </gazebo>

  <!-- MRL fingers -->
  <gazebo reference="middle_fle">
    <material>Gazebo/White</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>True</selfCollide>
  </gazebo>

  <gazebo reference="ring_fle">
    <material>Gazebo/White</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>True</selfCollide>
  </gazebo>

  <gazebo reference="little_fle">
    <material>Gazebo/White</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>True</selfCollide>
  </gazebo>

  <gazebo reference="j_index_fle">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>

  <gazebo reference="j_little_fle">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>

  <gazebo reference="j_mrl_fle">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>

  <gazebo reference="j_ring_fle">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>

  <gazebo reference="j_thumb_fle">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>

  <gazebo reference="j_thumb_opp">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>

</xacro:macro>
</robot>
